src/garage/torch/modules/gaussian_mlp_module.py

```diff
@@ -0,0 +1,255 @@
+    """
+    def __init__(self,
+                 input_dim,
+                 output_dim,
+                 **kwargs):
+        super(GaussianMLPModule, self).__init__(input_dim, output_dim, **kwargs)
+
+        self._mean_module = MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._hidden_sizes,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._output_b_init,
+            layer_normalization=self._layer_normalization)
+
+    def _get_mean_and_log_std(self, inputs):
+        mean = self._mean_module(inputs)
+
+        broadcast_shape = list(inputs.shape[:-1]) + [self._action_dim]
+        uncentered_log_std = torch.zeros(*broadcast_shape) + self._init_std_param
+
+        return mean, uncentered_log_std
+
+
+class GaussianMLPAdaptiveModule(GaussianMLPBaseModule):
+    """
+    GaussianMLPModule which has two different mean and std network.
+    """
+    def __init__(self,
+                 input_dim,
+                 output_dim,
+                 **kwargs):
+        super(GaussianMLPAdaptiveModule, self).__init__(input_dim, output_dim, **kwargs)
+
+        self._mean_module = MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._hidden_sizes,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._output_b_init,
+            layer_normalization=self._layer_normalization)
+
+        self._log_std_module = MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._std_hidden_sizes,
+            hidden_nonlinearity=self._std_hidden_nonlinearity,
+            hidden_w_init=self._std_hidden_w_init,
+            hidden_b_init=self._std_hidden_b_init,
+            output_nonlinearity=self._std_output_nonlinearity,
+            output_w_init=self._std_output_w_init,
+            output_b_init=self._init_adaptive_b,
+            layer_normalization=self._layer_normalization)
+
+    def _init_adaptive_b(self, b):
+        return nn.init.constant_(b, self._init_std_param.item())
+
+    def _get_mean_and_log_std(self, inputs):
+        return self._mean_module(inputs), self._log_std_module(inputs)
+
+
+class GaussianMLPSharedModule(GaussianMLPBaseModule):
+    """
+    GaussianMLPModule which has only one mean network.
+    Std will be a parameter
+    """
+    def __init__(self,
+                 input_dim,
+                 output_dim,
+                 **kwargs):
```

Comment:
please never use **kwargs except in very special code. all args you pass along should be declared.

Reply:
fixed